src/menu/views/system_info.c

```diff
@@ -75,11 +77,12 @@ static void draw (menu_t *menu, surface_t *d) {
 }
 
 
-void view_system_info_init (menu_t *menu) {    
-    controllers = get_controllers_present();
-
+void view_system_info_init (menu_t *menu) {
     for (int i = 0; i < 4; i++) {
-        accessory[i] = identify_accessory(i);
+        joypad[i] = joypad_is_connected(i);
+        if (joypad[i]) {
```

Comment:
There is no harm in calling `joypad_get_accessory_type` for disconnected ports. This check is not necessary.